app/controllers/concerns/form_submittable.rb

```diff
@@ -0,0 +1,167 @@
+  #
+  # The average use case will most likely only require to override slug-specific callbacks
+  # that are used to execute custom logic before, after, or around certain actions.
+  # In some cases, you may not need to define any callbacks at all.
+  #
+  # Default behaviour summary for each action:
+  #
+  #   controller#new: `redirect_to_first_slug`
+  #   controller#show: `before_show` -> `render_template_for_current_slug`
+  #   controller#update: `before_update` -> `handle_form_submission` ->
+  #     -> `form#save` succeded? ->
+  #       -> `execute_callback_if_exists(:after_form_save_success)` OR `redirect_to_next_slug`
+  #     -> `form#save failed?` ->
+  #       -> `execute_callback_if_exists(:after_form_save_failure)` OR `render_template_for_current_slug
+  #   controller#create: same as controller#update
+  #
+  # When including this concern make sure that these methods are accessible in the controller:
+  #
+  # - `slugs`, `current_slug`, `next_slug` (normally from `PageSequence`, or overridden)
+  # - `journey` (from `PartOfJourneyConcern`)
+  # - `current_data_object`, required to load the form object
+  # - any slug-specific callbacks (via a separate mixin)
+  #
+  # Important: `current_slug` is used to generate callbacks and assumed safe to use, i.e. derived
+  # from constrained, validated or sanitised user input.
+  #
+  # Important: If there are other callbacks in the controller, you should be including this concern
+  # **after** all the callbacks are defined. Placing this at the top is most likely a bad idea.
+  # In most cases, form submission callbacks are meant to kick in last.
+  #
+  # See the implementation of `_set_slug_specific_callbacks` below for more details.
+
+  included do
+    before_action :_set_slug_specific_callbacks, only: [:show, :update, :create]
+    before_action :before_show, only: :show
+    before_action :before_update, only: [:update, :create]
+    before_action :load_form_if_exists, only: [:show, :update, :create]
+    around_action :handle_form_submission, only: [:update, :create]
+
+    def current_data_object
+      # This is the instance of the main resource handled by the form object,
+      # and should always be defined in the controller.
+      nil
+    end
+
+    def new
+      redirect_to_first_slug
+    end
+
+    def show
+      render_template_for_current_slug
+    end
+
+    def create
+      # Note: if implemented, this action will be yielded at the end of `handle_form_submission`
+    end
+
+    def update
+      # Note: if implemented, this action will be yielded at the end of `handle_form_submission`
+    end
+
+    private
+
+    #
+    # Slug-specific callbacks are generated and executed around `show`, `update`, `create` actions,
+    # For example, for the "personal-details" slug, the following callbacks are available:
+    #
+    # `personal_details_before_show`
+    # `personal_details_before_update`
+    # `personal_details_after_form_save_success` (*)
```

Comment:
I think we will want to iterate towards the forms _not_ being responsible for persistence, therefore moving those responsibilities to one of these callbacks might make sense. Given that this is likely to be repeated on everyt form I don't mind if we decide to keep as-is for simplicity, but I suspect we may find a way to do this in a re-usable way when we don't have the unnecessary complexity of multiple policy `Claim` records and `::Eligibility` records to work with. I would imagine we would only need a very simple interface to save the form's attributes to the `Journey:Session` record in this new `handle_form_submission` method. 

The naming of `after_form_save_success` might be inappropriate in that case. Whether we decide to change it now or later I don't mind, but I anticipate we would change it fairly soon.

Reply:
I see what you are saying; we can try to do that in the next iteration